app/models/user.rb

```diff
@@ -1,20 +1,27 @@
 class User < ApplicationRecord
   # Include default devise modules. Others available are:
-  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  # :lockable, :timeoutable, :trackable and :omniauthable
   devise :database_authenticatable, :registerable,
-         :recoverable, :rememberable, :validatable
-         has_many :cars
-
-  attr_writer :login
-  validate :validate_username
-  validates_uniqueness_of :username
-  has_rich_text :content
-  validate :content
+         :recoverable, :rememberable, :validatable, :confirmable
+   
+  attr_writer :login       
+  has_many :cars  
+  validate :validate_username, :content
+  validates_uniqueness_of :username, :email
+  has_one_attached :content do |attachable|
+    attachable.variant :thumb, resize_to_limit: [46, 46]
+  end
+  validates :email, format: { with: /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i, on: :create }, length: {maximum: 320} 
+  validates :username, presence: true, length: {maximum: 40} 
+  validates :password, length: {maximum: 20}  
+  validate :validate_filetypes
```

Comment:
I did make callbacks for username length and for type of fails , but couldn't make it work for the email and password

Reply:
Oh my god, I'm sorry if I selected 4 fields at once. I meant to create a callback instead of :validate_filetypes because we usually validate only the table fields. Lines 14-16 we perfectly fine

Reply:
No problem at all, I like to learn things in hard way, that's how I remember them better 🤣